symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/MessageService.java

```diff
@@ -357,11 +369,15 @@ public MessageReceiptDetailResponse listMessageReceipts(@Nonnull String messageI
    * @see <a href="https://developers.symphony.com/restapi/reference#message-metadata-relationship">Message Metadata</a>
    */
   public MessageMetadataResponse getMessageRelationships(@Nonnull String messageId) {
-    return callAndCatchApiException(() -> defaultApi.v1AdminMessagesMessageIdMetadataRelationshipsGet(
+    return executeAndRetry("getMessageRelationships", () -> defaultApi.v1AdminMessagesMessageIdMetadataRelationshipsGet(
         authSession.getSessionToken(), ApiUtils.getUserAgent(), messageId));
   }
 
   private static Long getEpochMillis(Instant instant) {
     return instant == null ? null : instant.toEpochMilli();
   }
+
+  private <T> T executeAndRetry(String name, SupplierWithApiException<T> supplier) {
+    return RetryWithRecovery.executeAndRetry(retryBuilder, name, supplier);
```

Comment:
Could we have `this.retryBuilder.execute(name, supplier)`?

Reply:
I just realized it won't work because we use a same base `retryBuilder` for suppliers returning different types. So we have to keep it static with a param type which can be specified

Reply:
Understood 👍